clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -167,7 +167,7 @@
  * called <i>test</i> as described above.
  * <p>
  * The broker will automatically detect failed processes in the <i>test</i> group by using a heartbeat mechanism. The
- * consumer will automatically ping the cluster periodically, which lets the cluster know that it is alive. As long as
+ * consumer will ping the cluster whenever {@link #poll(long)} is called, which lets the cluster know that it is alive. As long as
```

Comment:
That's a good clarification.  How about this for the whole paragraph: 

> The broker will automatically detect failed processes in the <i>test</i> group by using a heartbeat mechanism. The consumer will automatically ping the cluster periodically, which lets the cluster know that it is alive. Note that the consumer is single-threaded, so periodic heartbeats can only be sent when poll() is called. As long as the consumer is able to do this it is considered alive and retains the right to consume from the partitions assigned to it. If it stops heartbeating or fails to call poll() for a period of time longer than <code>session.timeout.ms</code> then it will be considered dead and its partitions will be assigned to another process.